Added MultiLineString support for @turf/point-on-line

stebogit opened the pull request:
Fixes #835 

Still the module returns a single `point`, the closest of all the `lines` of the `MultiLineString`.
Is it correct?

CC: @jbccollins @DenisCarriere

DenisCarriere commented on packages/turf-point-on-line/index.js:
```diff
@@ -44,24 +47,22 @@ var lineString = helpers.lineString;
-        throw new Error('input must be a LineString Feature or Geometry');
+module.exports = function (lines, pt, units) {
+    // validation
+    if (lines.type !== 'Feature' && lines.type !== 'LineString' && lines.type !== 'MultiLineString' && lines.geometry.type !== 'LineString' && lines.geometry.type !== 'MultiLineString') {
```
Condensed this validation step to this:

```js
var type = (lines.geometry) ? lines.geometry.type : lines.type;
if (type !== 'LineString' && type !== 'MultiLineString') {
    throw new Error('lines must be LineString or MultiLineString');
}
```

stebogit replied:
🤦‍♂️ of course! Dumb me 😅

DenisCarriere replied:
No no... it's just simpler to read this way.

stebogit replied:
exactly!
simpler => better!!  😃

DenisCarriere replied:
Agreed! 😄

DenisCarriere commented on packages/turf-point-on-line/index.js:
```diff
@@ -1,18 +1,21 @@
  *
  * @name pointOnLine
- * @param {Feature<LineString>} line line to snap to
+ * @param {Feature<LineString>|Feature<MultiLineString>} lines lines to snap to
```
To support both Feature & Geometry, so far the JSDocs have been declared like this:

```js
/**
 * @param {Geometry|Feature<LineString|MultiLineString>} lines lines to snap to
 * @param {Geometry|Feature<Point>|number[]} pt point to snap from
```

DenisCarriere commented:
Using happening when the point is between the MultiLineString.
![image](https://user-images.githubusercontent.com/550895/28124019-7dbbb6de-66f0-11e7-840e-306a54090f49.png)

stebogit commented:
Fixed with `flattenEach`.

![screen shot 2017-07-12 at 11 52 17 pm](https://user-images.githubusercontent.com/12717225/28154053-7bc214dc-675d-11e7-9711-31da65ae10b3.png)

Added also a more complex test:
![screen shot 2017-07-12 at 11 58 07 pm](https://user-images.githubusercontent.com/12717225/28154190-f9d82154-675d-11e7-9284-587f245c46b9.png)

DenisCarriere commented:
@stebogit Awesome! Looks good to me
